normandy/recipes/api/views.py

```diff
@@ -155,6 +144,101 @@ def disable(self, request, pk=None):
         recipe.disable()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @detail_route(methods=['GET'])
+    def approval_requests(self, request, pk=None):
+        recipe = self.get_object()
+        serializer = ApprovalRequestSerializer(recipe.approval_requests, many=True)
+        return Response(serializer.data)
+
+
+class ApprovalRequestViewSet(viewsets.ModelViewSet):
+    queryset = ApprovalRequest.objects.all()
+    serializer_class = ApprovalRequestSerializer
+    filter_fields = ('active',)
+    permission_classes = [
+        permissions.DjangoModelPermissionsOrAnonReadOnly,
+        AdminEnabledOrReadOnly,
+    ]
+
+    @transaction.atomic()
+    @reversion.create_revision()
+    def create(self, request, *args, **kwargs):
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError:
```

Comment:
Actually I lied before; we should have a custom exception for this, and then catch that in `dispatch` so that we don't have to repeatedly catch this exception everywhere.